### PR TITLE
Arduino NimBLE support

### PR DESCRIPTION
**Pull request by 1technophile**
Arduino NIMBLE support (less flash and memory usage, better stability) 
Decorrelate the device discovery from the BLE scan so as to avoid canary stack trace error and prepare the future add of BLEconnect

### CODE REVIEW

File: main/ZgatewayBT.ino [hunk 1]
```diff
@@ -430,12 +437,11 @@ void BLEscan() {
   Log.notice(F("Scan begin" CR));
   BLEDevice::init("");
   BLEScan* pBLEScan = BLEDevice::getScan(); //create new scan
-  MyAdvertisedDeviceCallbacks myCallbacks;
-  pBLEScan->setAdvertisedDeviceCallbacks(&myCallbacks);
+  pBLEScan->setAdvertisedDeviceCallbacks(new MyAdvertisedDeviceCallbacks());
```
Comment (Legion2) [inline main/ZgatewayBT.ino:440]: Is the allocated memory later deallocated?

Comment (1technophile) [reply main/ZgatewayBT.ino:440]: Yes by 
`BLEDevice::deinit(true);`
usage
https://github.com/h2zero/NimBLE-Arduino/blob/d8a535d28ab7af8a4074f719fb87406c4b244a69/src/NimBLEDevice.cpp#L543

**Comment by 1technophile** [comment]
Thanks @jmw6773 

@Legion2 do you have other comments for this PR?

**Comment by Legion2** [comment]
@1technophile no I only looked at the code, I did not test it, I have no bluetooth devices.

**Comment by 1technophile** [comment]
ok, thanks!

**Pull request merged**